From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: aspect <aspect@anthropic.com>
Date: Tue, 10 Dec 2024 00:00:00 +0000
Subject: [PATCH] Fix XKB context creation for Android

Use XKB_CONTEXT_NO_DEFAULT_INCLUDES instead of XKB_CONTEXT_NO_FLAGS
when creating the XKB context. This prevents xkbcommon from trying
to load configuration files from paths that don't exist on Android
(e.g., /usr/share/X11/xkb), which would cause xkb_context_new() to
return NULL and lead to a crash in xkb_context_ref().
---
 Source/WebKit/WPEPlatform/wpe/WPEKeymapXKB.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Source/WebKit/WPEPlatform/wpe/WPEKeymapXKB.cpp b/Source/WebKit/WPEPlatform/wpe/WPEKeymapXKB.cpp
index 1111111111..2222222222 100644
--- a/Source/WebKit/WPEPlatform/wpe/WPEKeymapXKB.cpp
+++ b/Source/WebKit/WPEPlatform/wpe/WPEKeymapXKB.cpp
@@ -176,7 +176,7 @@ WPEKeymap* wpe_keymap_xkb_new()
 {
     auto* keymap = WPE_KEYMAP_XKB(g_object_new(WPE_TYPE_KEYMAP_XKB, nullptr));

-    struct xkb_context* context = xkb_context_new(XKB_CONTEXT_NO_FLAGS);
+    struct xkb_context* context = xkb_context_new(XKB_CONTEXT_NO_DEFAULT_INCLUDES);
     struct xkb_rule_names names = { "evdev", "pc105", "us", "", "" };
     keymap->priv->xkbKeymap = xkb_keymap_new_from_names(context, &names, XKB_KEYMAP_COMPILE_NO_FLAGS);
     keymap->priv->xkbState = xkb_state_new(keymap->priv->xkbKeymap);
--
2.43.0
